### PR TITLE
68 project gallery

### PR DESCRIPTION
**Pull request by ryanyae**
## Description

Created a new component that is a carousel for projects

## Updates

created a new component and added only instance of it in the Correlation's project

## Checklist

- [x] Ran `npm run lint:fix` to format code
- [x] Requested at least one reviewer
- [x] Connected PR to Trello ticket (steps below)
- [ ] Addressed code review comments
- [ ] Gained at least one approval for your PR

## Connecting your PR to the corresponding ticket:
- Click on the "GitHub" button found on the right of your Trello ticket
- Choose "Attach PR" from the dropdown list
- Link your GitHub account (if you haven't done so already)
- Navigate to`VCL-content-platform` and choose your PR from the dropdown.

### CODE REVIEW

**Comment by ryanyae** [comment]
[Project Gallery](https://trello.com/c/C0xaQdw5/68-project-gallery)

**Comment by ryanyae** [comment]
This is one of the questions that I had, I had tried to make a 3D carousel and noticed that there was a lot of hardcode and ended up with this implementation. If this is not the type of carousel that is desired I could try and go implement a 3D version of it.

**Comment by ElisaHu1** [comment]
image
instead of changing the package.lock.json, you should change pakage.json instead for this PR

**Review by ElisaHu1 — COMMENTED** [review]
looks amazing, are you able to change the scale of the image and leave some paddings like the pic in figma
image
so the images does not look distorted

**Comment by ryanyae** [comment]
@ElisaHu1, right I for sure am able to. Earliest I can get you these changes maybe tomorrow afternoon.

File: package.json [hunk 1]
```diff
@@ -37,7 +38,7 @@
     "yup": "^0.32.9"
   },
   "scripts": {
-    "start": "export SET NODE_OPTIONS=--openssl-legacy-provider && react-app-rewired start",
+    "start": "set NODE_OPTIONS=--openssl-legacy-provider && react-app-rewired start",
```
Comment (svolterra) [inline package.json:41]: Hmm interesting that package.json and package-lock.json are both in the .gitignore file yet they were still committed 🤔

File: src/components/ProjectGallery/ProjectGallery.tsx [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    const timeoutRef = useRef<any>(null)
+
+    // Will just restart the timer within the Timeout obj
+    const resetTimeout = () => {
+        if (timeoutRef.current) {
+            clearTimeout(timeoutRef.current)
+        }
+    }
+
+    // This is the autoscrolling feature
```
Comment (svolterra) [inline src/components/ProjectGallery/ProjectGallery.tsx:61]: Amazing addition! Works really well too :) Great work!

File: src/components/ProjectGallery/ProjectGallery.tsx [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+        }
+    }, [galleryIndex])
+
+    return (
+        <>
+            <div id='container'>
+                <div id='container-titles'>
+                    {
+                        /* 
+                            The titles below should not be hardcoded like this, but should be instead inside a prop
```
Comment (svolterra) [inline src/components/ProjectGallery/ProjectGallery.tsx:78]: 100% agree with this comment :)

**Review by svolterra — COMMENTED** [review]
Really really great work Ryan! The gallery looks incredible and I absolutely love the auto-scrolling feature 😁 I do agree with Elisa on the photo padding, but otherwise looks amazing :) Phenomenal work :)

**Review by svolterra — COMMENTED** [review]
Thank you for addressing the code review comments! I was just wondering if it would be possible to move the image title and the image itself slightly to the right such that their positions align more with the **Gallery** title? They appear to be a little off-centered at the moment 🤔 

Screen Shot 2023-07-03 at 8 43 30 PM

Otherwise looks amazing 💯 🏆 👏

**Comment by ryanyae** [comment]
> Thank you for addressing the code review comments! I was just wondering if it would be possible to move the image title and the image itself slightly to the right such that their positions align more with the **Gallery** title? They appear to be a little off-centered at the moment 🤔
> 
> Screen Shot 2023-07-03 at 8 43 30 PM
> Otherwise looks amazing 💯 🏆 👏

I just fixed it with my most recent push. It was just a little CSS component that needed to be fixed

File: package.json (hunk 1 above)
Comment (ryanyae) [reply package.json]: Yea, I am kind of struggling with this fix. Been searching on how to fix this problem and none seem work. I will try to figure out a fix to this problem, hopefully soon.

**Review by ElisaHu1 — APPROVED** [review]
autoscrolling stoped working with the new commit, besides that, I am good with this pr

**Comment by ryanyae** [comment]
@ElisaHu1 @svolterra I made changes to the branch after I fixed what Elisa had mentioned and I do not see the commit here. I hope they fix by itself, but if it doesn't I could bring it up in tomorrows meeting.

**Pull request merged**